x-pack/plugins/security_solution/public/resolver/store/data/reducer.test.ts

```diff
@@ -22,14 +23,46 @@ describe('Resolver Data Middleware', () => {
   });
 
   describe('when data was received and the ancestry and children edges had cursors', () => {
+    let firstChildNodeInTree: TreeNode;
+    let statsForFirstChild: Record<string, number>;
     beforeEach(() => {
       const generator = new EndpointDocGenerator('seed');
+      const baseTree = generator.generateTree({
+        ancestors: 1,
+        generations: 2,
+        children: 3,
+        percentWithRelated: 100,
+        alwaysGenMaxChildrenPerNode: true,
+      });
+      const { children } = baseTree;
+      firstChildNodeInTree = [...children.values()][0];
+      statsForFirstChild = firstChildNodeInTree.relatedEvents.reduce(
+        (counts: Record<string, number>, relatedEvent) => {
+          for (const category of [relatedEvent.event.category].flat()) {
+            counts[category] = counts[category] ? counts[category] + 1 : 1;
+          }
+          return counts;
+        },
+        {}
+      );
       const tree = mockResolverTree({
-        events: generator.generateTree({ ancestors: 1, generations: 2, children: 2 }).allEvents,
+        events: baseTree.allEvents,
         cursors: {
-          childrenNextChild: 'aValidChildursor',
+          childrenNextChild: 'aValidChildCursor',
           ancestryNextAncestor: 'aValidAncestorCursor',
         },
+        children: [...baseTree.children.values()].map((node) => {
+          const childNode: Partial<ResolverChildNode> = node;
+          childNode.entityID = node.id;
+          if (node.id === firstChildNodeInTree.id) {
+            // attach stats
+            childNode.stats = {
+              events: { total: 1, byCategory: statsForFirstChild },
+              totalAlerts: 0,
```

Comment:
is it important that `totalAlerts` is 0? does that match the data in the mock?

Reply:
No, but TS requires it.